lib/api/CAnomalyJob.cc

```diff
@@ -1051,15 +1055,26 @@ bool CAnomalyJob::backgroundPersistState(CBackgroundPersister& backgroundPersist
     std::sort(copiedDetectors.begin(), copiedDetectors.end(),
               maths::COrderings::SFirstLess());
 
-    if (backgroundPersister.addPersistFunc(std::bind(
+    if (m_PeriodicPersister->addPersistFunc(std::bind(
             &CAnomalyJob::runBackgroundPersist, this, args, std::placeholders::_1)) == false) {
         LOG_ERROR(<< "Failed to add anomaly detector background persistence function");
         return false;
     }
 
+    m_PeriodicPersister->persistInForeground(false);
```

Comment:
When I first saw this line I thought it was a bug.  I think it's just that the naming is confusing.  If I'm reading this correctly `persistInForeground(false)` means persist in the background.  Maybe `runPersistFuncs(false)` would be less confusing.  Or maybe there is a name that is better still.  But having `Foreground` in the name seems wrong.

Reply:
There's acres of debate around boolean parameters on the internet. 

https://martinfowler.com/bliki/FlagArgument.html